src/coreclr/gc/gcee.cpp

```diff
@@ -165,6 +177,52 @@ void GCHeap::UpdatePostGCCounters()
 
     FIRE_EVENT(GCEnd_V1, static_cast<uint32_t>(pSettings->gc_index), condemned_gen);
 
+    // Fire custom event for process private memory at GC end
```

Comment:
Should this be done only when there are listeners for this event? Collecting this data is not cheap, e.g. it requires opening and closing a file on Linux and parsing its content.